container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/logs_aggregator_functions/package_private_helpers.go

```diff
@@ -0,0 +1,105 @@
+package logs_aggregator_functions
+
+import (
+	"context"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/consts"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/docker/docker_manager"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/docker/docker_manager/types"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/docker/object_attributes_provider/label_key_consts"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/docker/object_attributes_provider/label_value_consts"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/container_status"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/logs_aggregator"
+	"github.com/kurtosis-tech/stacktrace"
+	"net"
+)
+
+const (
+	shouldShowStoppedLogsAggregatorContainers = true
+)
+
+func getLogsAggregatorObjectAndContainerId(
+	ctx context.Context,
+	dockerManager *docker_manager.DockerManager,
+) (
+	resultMaybeLogsAggregator *logs_aggregator.LogsAggregator,
+	resultMaybeContainerId string,
+	resultErr error,
+) {
+	allLogsAggregatorContainers, err := getAllLogsAggregatorContainers(ctx, dockerManager)
+	if err != nil {
+		return nil, "", stacktrace.Propagate(err, "An error occurred getting all logs Aggregator containers")
+	}
+
+	if len(allLogsAggregatorContainers) == 0 {
+		return nil, "", nil
+	}
+	if len(allLogsAggregatorContainers) > 1 {
+		return nil, "", stacktrace.NewError("Found more than one logs Aggregator Docker container'; this is a bug in Kurtosis")
+	}
+
+	logsAggregatorContainer := allLogsAggregatorContainers[0]
+	logsAggregatorContainerID := logsAggregatorContainer.GetId()
+
+	logsAggregatorObject, err := getLogsAggregatorObjectFromContainerInfo(
+		ctx,
+		logsAggregatorContainerID,
+		logsAggregatorContainer.GetStatus(),
+		dockerManager,
+	)
+	if err != nil {
+		return nil, "", stacktrace.Propagate(err, "An error occurred getting the logs Aggregator object using container ID '%v', labels '%+v' and the status '%v'", logsAggregatorContainer.GetId(), logsAggregatorContainer.GetLabels(), logsAggregatorContainer.GetStatus())
+	}
+
+	return logsAggregatorObject, logsAggregatorContainerID, nil
+}
+
+func getAllLogsAggregatorContainers(ctx context.Context, dockerManager *docker_manager.DockerManager) ([]*types.Container, error) {
```

Comment:
Can there be more than one log aggregator? I'm confused. If not, can we make this function `getLogsAggregatorContainer` and throw if there's more than one? Or do you account for stopped ones as well?

Reply:
yes should only be one log aggregator, fixed